addons/hr_holidays/static/src/js/time_off_calendar.js

```diff
@@ -138,6 +138,27 @@ odoo.define('hr_holidays.dashboard.view_custo', function(require) {
             var self = this;
             return this._super.apply(this, arguments).then(function () {
                 self.$el.parent().find('.o_calendar_mini').hide();
+
+                // Check if there is a filter to display on the sidebar
+                // If there is no filter, hide the sidebar
+                var no_filters = true;
+                for (var filter in self.state.filters) {
+                    if (self.state.filters[filter].filters && self.state.filters[filter].filters.length > 0) {
+                        no_filters = false;
+                        break;
+                    }
+                }
+
+                // Remove the no data sidebar
+                self.$sidebar.find('#o_calendar_filter_no_data').remove();
+                if (no_filters) {
+                    // Show a special sidebar
+                    self.$sidebar.html(QWeb.render('hr_holidays.calendar.sidebar.nofilter', {
+                        title: 'Time Off Type',
+                        description: '(no data)'
+                    }));
+                }
+                return Promise.resolve();
```

Comment:
not mandatory unless you want to add a 1 tick delay which is I think useless here